CompileIO/compile.io
# Run tests backend 26

# Addressed Issue(s)

#26 

# Goals

Let's us run JUnit tests on the back-end for uploaded student code.

# Special Considerations

Only works for Java, multi-file not configured for python. The proper `build.gradle` file needs to be in the `upload-directory` (wherever that may be). 

# Acceptance Criteria

JUnit tests can now run against submitted java files within a Docker container.

# Testing
**TO TEST INDEPENDENT OF FRONT-END**
- Check slack `team-chat` for proper files needed, like the build.gradle file.
- Check `Main.java` for an example on how to run the back-end independent of the front-end.
- Move the `build.gradle` file to the `upload-directory` you will be using for manual tests (if this is confusing please contact me)
- To run unit tests, navigate to `compile.io/Spring_Server/gs-uploading-files-complete` and run `gradle test`. Comprehensive test results can be viewed in `gs-uploading.../build/reports/tests/test/index.html`

# Branch

All work was done on `run-tests-backend-26`
